plugins/module_utils/foreman_helper.py

```diff
@@ -829,13 +901,41 @@ def run(self, module_params=None, entity=None, search=None):
             Returns entity.
         """
         if not module_params:
-            module_params = self.clean_params()
+            module_params = self.foreman_params
 
         entity, module_params = self.resolve_entities(module_params, entity, search)
 
         params = {}
         if self.entity_scope:
-            params['{0}_id'.format(self.entity_scope)] = module_params[self.entity_scope]['id']
+            params['{0}_id'.format(self.entity_scope[0])] = module_params[self.entity_scope[0]]['id']
+        new_entity = self.ensure_entity(self._entity_resource_name, module_params, entity, params=params)
+        new_entity = self.remove_sensitive_fields(new_entity)
+
+        if new_entity and 'parameters' in module_params:
+            if 'parameters' in self.foreman_spec and self.foreman_spec['parameters'].get('type') == 'nested_list':
+                scope = {'{0}_id'.format(self.entity_name): new_entity['id']}
+                self.ensure_scoped_parameters(scope, entity, module_params.get('parameters'))
+
+        return new_entity
+
+    def cycle(self):
+        """ lookup entities, ensure entity, remove sensitive data, manage parameters.
+            Like 'run', just faster and more convenient...
+        """
+        module_params = self.foreman_params
+        if not self.desired_absent:
+            self.auto_lookup_entities()
```

Comment:
I keep thinking, that we would not even need this, if we did the entity lookup completely lazy.